Tidy __all__ under torch._refs

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103712

- Added ops that were missing under `__all__`.
- Some misc changes to helper functions to make them private.
- Set correct `fn.__module__` for `fn` created by `_make_alias`, when called in another module.

All modification largely references results from a hacked version of `test_public_bindings::test_correct_module_names`.
By default `torch._refs` is not included in the test because it is technically a private package.

cc @ezyang @mruberry @Lezcano @peterbell10

BowenBao committed:
Tidy __all__ under torch._refs

[ghstack-poisoned]

BowenBao committed:
Add skips for new discovered ops on "Tidy __all__ under torch._refs"


- Added ops that were missing under `__all__`.
- Some misc changes to helper functions to make them private.
- Set correct `fn.__module__` for `fn` created by `_make_alias`, when called in another module.

All modification largely references results from a hacked version of `test_public_bindings::test_correct_module_names`.
By default `torch._refs` is not included in the test because it is technically a private package.

[ghstack-poisoned]

lezcano commented on torch/_refs/nn/functional/__init__.py:
```diff
@@ -116,7 +125,7 @@ def alpha_dropout(
 
 
-def inplace_wrapper(fn):
+def _inplace_wrapper(fn):
```
We already have `out_wrapper` without an initial underscore, so this one can probably be without an underscore as well.

BowenBao replied:
I think a closer comparison is to `_make_inplace` under `torch._refs.__init__.py`. IMO both are module specific helper function so I'd prefer not exposing publicly. wdyt?

lezcano commented on torch/_refs/__init__.py:
```diff
@@ -415,16 +439,19 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
         return fn(*args, **kwargs)
 
     _fn.__name__ = name
+    _fn.__module__ = module_name
```
If you want to do this automatically, you can inspect at the module fo the previous frame. It'd allow for a neater and less error-prone API at the cost of one magic trick.

BowenBao replied:
okay, I was originally thinking the same but worried it would be considered too hacky.

BowenBao committed:
inspect module name on "Tidy __all__ under torch._refs"


- Added ops that were missing under `__all__`.
- Some misc changes to helper functions to make them private.
- Set correct `fn.__module__` for `fn` created by `_make_alias`, when called in another module.

All modification largely references results from a hacked version of `test_public_bindings::test_correct_module_names`.
By default `torch._refs` is not included in the test because it is technically a private package.

cc ezyang mruberry Lezcano peterbell10

[ghstack-poisoned]

BowenBao commented:
Thanks for review @lezcano ! Updated according to comments.

lezcano reviewed (approved):
Cool, thanks!

BowenBao committed:
lint on "Tidy __all__ under torch._refs"


- Added ops that were missing under `__all__`.
- Some misc changes to helper functions to make them private.
- Set correct `fn.__module__` for `fn` created by `_make_alias`, when called in another module.

All modification largely references results from a hacked version of `test_public_bindings::test_correct_module_names`.
By default `torch._refs` is not included in the test because it is technically a private package.

cc ezyang mruberry Lezcano peterbell10

[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge

BowenBao commented:
@pytorchbot merge
